pkg/apis/postgres-operator.crunchydata.com/README.md

```diff
@@ -0,0 +1,121 @@
+<!--
+# Copyright 2025 Crunchy Data Solutions, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+-->
+
+# Custom Resource Definitions
+
+These directories contain Go types that serve as [DTO]s for communicating with the [Kubernetes API].
+We use [controller-gen] to produce [CRD]s based on these Go types with [schemas](validation.md) that match.
+
+This [directory](.) contains our API Group, `postgres-operator.crunchydata.com`, and each subdirectory is a version:
+
+- v1beta1 is compatible with Kubernetes 1.30, OpenShift 4.14, and later
+- v1 uses newer CRD features and requires Kubernetes 1.30, OpenShift 4.17, and later
+
+```
+pkg/apis/postgres-operator.crunchydata.com
+├── v1
+└── v1beta1
+```
+
+[controller-gen]: https://book.kubebuilder.io/reference/controller-gen
+[CRD]: https://docs.k8s.io/tasks/extend-kubernetes/custom-resources/custom-resource-definitions
+[DTO]: https://martinfowler.com/eaaCatalog/dataTransferObject.html
+[Kubernetes API]: https://docs.k8s.io/concepts/overview/kubernetes-api
+
+
+# CRD Versions
+
+Kubernetes organizes API resources into Groups. Each resource is represented by a Kind that can have multiple Versions. The shape of a CRD reflects this:
+
+```yaml
+kind: CustomResourceDefinition
+metadata:
+  name: "ideas.example.com" # {spec.plural}.{spec.group}
+spec:
+  group: "example.com" # one group (G)
+  names:
+    kind: Idea         # one kind (K)
+    plural: ideas      # one resource (R)
+    singular: idea     # one resource (R)
+  versions:            # many versions (V)
+    - name: v1beta1
+      schema: …
+    - name: v1
+      schema: …
+```
+
+<!--
+```mermaid
```

Comment:
<img width="1212" height="470" alt="Screenshot 2025-10-01 at 2 01 41 PM" src="https://github.com/user-attachments/assets/2bca11e9-cde4-49d8-90bf-6ee39b649294" />

Reply:
🤷🏻 I wasn't sure how helpful this diagram would be, so I left it in _and_ out simultaneously. 😁